redaxo/src/core/lib/util/socket/socket_response.php

```diff
@@ -209,13 +238,80 @@ public function getBody()
         if (null === $this->body) {
             $this->body = '';
 
+            $appendedZlibStreamFilter = null;
+
+            // Decode the content for gzip and deflate
+            if ($this->isGzipOrDeflateEncoded() && $this->decompressContent && !$this->chunked) {
+                $appendedZlibStreamFilter = $this->addZlibStreamFilter($this->stream, STREAM_FILTER_READ);
+            }
+
             while (false !== ($buf = $this->getBufferedBody())) {
                 $this->body .= $buf;
             }
+
+            if ($this->isGzipOrDeflateEncoded() && $this->decompressContent && $this->chunked) {
+                $stream = fopen('php://temp', 'r+');
+                fwrite($stream, $this->body);
+                fseek($stream, 0);
```

Comment:
alternativ geht vllt sowas

http://php.adamharvey.name/manual/de/function.http-chunked-decode.php

Reply:
Wenn man einfach nur mit getBufferedBody arbeitet (von außen, ist ja public), greift dann das Dekodieren schon? Müsste nicht eher direkt in der Methode auch was passieren?